src/React.Core/JavaScriptEngineFactory.cs

```diff
@@ -233,13 +233,13 @@ public virtual IJsEngine GetEngine()
 		/// Returns an engine to the pool so it can be reused
 		/// </summary>
 		/// <param name="engine">Engine to return</param>
-		public virtual void ReturnEngineToPool(IJsEngine engine)
+		public virtual void ReturnEngineToPool(PooledJsEngine engine)
 		{
 			// This could be called from ReactEnvironment.Dispose if that class is disposed after 
 			// this class. Let's just ignore this if it's disposed.
 			if (!_disposed)
 			{
-				_pool.ReturnEngineToPool(engine);	
+				engine.Dispose();
```

Comment:
This is probably not needed... The call site could just call `Dispose` on the engine directly.